git-subhistory.sh

```diff
@@ -237,23 +242,24 @@ subhistory_assimilate () {
 					git read-tree $(
 						cat "$GIT_DIR/subhistory-tmp/split-to-orig-map/$parent" 2>/dev/null \
 						|| map $parent) &&
-					git rm --cached -r '"'$path_to_sub'"' -q &&
+					git rm --cached -r --ignore-unmatch '"'$path_to_sub'"' -q &&
 					if test -z $parent_Main_tree
 					then
 						parent_Main_tree=$(git write-tree)
 					elif test $(git write-tree) != $parent_Main_tree
 					then
 						git read-tree '$default_Main_tree' && # TODO: find earliest merged tree
-						git rm --cached -r '"'$path_to_sub'"' -q &&
+						git rm --cached -r --ignore-unmatch '"'$path_to_sub'"' -q &&
 						break
 					fi
 				done
-			else
+			elif git rev-parse --verify -q $GIT_COMMIT^ >/dev/null # if this rev has a parent (IE is not a root/initial commit)
+                        then
```

Comment:
It was happening when testing with my repositories. See my general comments below for the major issue I'm looking at that I think led me to this.  In practice this will happen when you merge in a tree that brings in a new root commit (with no parents).